File: 프로그래머스/lv1/12943. 콜라츠 추측/콜라츠 추측.py
def solution(num):
    answer = 0
    
    while (num != 1) & (answer < 500):
        if num % 2 == 0:
            num /= 2
        else:
            num = num * 3 + 1
        answer += 1
    
    if (answer == 500) and (answer != 1):
        answer = -1
        
    return answer

File: 프로그래머스/1/12943. 콜라츠 추측/콜라츠 추측.java
class Solution {
    public int solution(int start) {
        int answer = 0;
        
        long num = (long)start;
        
        while (true) {
            if (num == 1) {
                break;
            }
            
            if (num % 2 == 0) {
                num /= 2;
            } else {
                num = (num * 3) + 1;
            }
            
            answer++;
            
            
            if (answer == 500) {
                answer = -1;
                break;
            }
            
        }
        
        return answer;
    }
}